lib/pool.py

```diff
@@ -185,3 +185,24 @@ def closeall(self):
             self._closeall()
         finally:
             self._lock.release()
+
+class MultiProcessThreadedConnectionPool(ThreadedConnectionPool):
+
+    def __init__(self, *args, **kwargs):
+        self.pid = os.getpid()
+        self.args = args
+        self.kwargs = kwargs
+        super().__init__(*args, **kwargs)
+
+    def getconn(self, *args, **kwargs):
+        current_pid = os.getpid()
+        if current_pid != self.pid:
+            super().__init__(*self.args, **self.kwargs)
```

Comment:
It's not good to just call `__init__` on a class already initialised. I think if you detect this state you should empty the pool: you could add a protected `_flush_pool()` method on the base class to encapsulate it.